### PR TITLE
`System::title()` and use it for email challenges

### PR DESCRIPTION
**Pull request by distantnative**
## Status: ready for review 👀

- The implementation of `System::title()` differs from https://github.com/getkirby/kirby/blob/b6cad90/config/api/models/System.php#L55-L61 as that code seemed flawed: the `catch` part could never be reached since `$site->blueprint()` always adds a fallback that is loaded with a title. Even more, the content value should be more important than the blueprint. That's fixed now.

### Enhancements
- Login emails (code and password reset) now include the site's title instead of a reference to Kirby

## Related issues/ideas

- Fixes https://github.com/getkirby/kirby/issues/3343

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [ ] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [x] Add changes to release notes draft in Notion

### CODE REVIEW

File: config/api/models/System.php [hunk 1]
```diff
@@ -52,12 +52,8 @@
...
             return $system->toArray();
         },
-        'site' => function () {
-            try {
-                return $this->site()->blueprint()->title();
-            } catch (Throwable $e) {
-                return $this->site()->title()->value();
-            }
+        'site' => function (System $system) {
+            return $system->title();
```
Comment (lukasbestle) [inline config/api/models/System.php:56]: I think it's weird that we now have a `site` field that returns the content title with blueprint fallback and a `title` field that only returns the content title. Can we unify this somehow?

To be honest I don't fully understand what the `site` and `title` are used for respectively.

File: src/Cms/System.php [hunk 2]
```diff
@@ -514,6 +514,18 @@ public function sessions(): bool
         return is_writable($this->app->root('sessions'));
     }
 
+    /**
+     * Return's the site's title as defined in the
+     * context file or `site.yml` blueprint
```
Comment (lukasbestle) [inline src/Cms/System.php]:
```suggestion
     * Returns the site's title as defined in the
     * content file or `site.yml` blueprint
```

File: config/api/models/System.php (hunk 1 above)
Comment (distantnative) [reply config/api/models/System.php:56]: I think we used it in different ways in the Panel. I don't think we use these anymore with Fiber,  but I also didn't want to remove them from the API as another breaking change.

Comment (lukasbestle) [reply config/api/models/System.php:56]: IMO we should check this and deprecate one of them, ideally `site`. Keeping both around is really confusing.

Comment (distantnative) [reply config/api/models/System.php:56]: As far as I can tell, on the JS we only use `$api.system.install()` from the system API  - `get()` and `register()` are completely unused. So from the Panel's side, we can do pretty much what we want in the API System model.

Comment (distantnative) [reply config/api/models/System.php:56]: We might even move the `install()` call to Fiber as well and consider deprecating the whole System part of the API.

But maybe we move that to its own issue and leave it here as is.

Comment (lukasbestle) [reply config/api/models/System.php:56]: Your call, it's only something I noticed and where I feel like we shouldn't leave it like that in the long term.

Comment (distantnative) [reply config/api/models/System.php:56]: https://github.com/getkirby/kirby/issues/3541

File: src/Cms/System.php [hunk 3]
```diff
@@ -514,6 +514,18 @@ public function sessions(): bool
...
+    /**
+     * Returns the site's title as defined in the
+     * content file or `site.yml` blueprint
+     *
+     * @return string
+     */
+    public function title(): string
+    {
+        $blueprint = $this->app->site()->blueprint()->title();
+        return $this->app->site()->title()->or($blueprint)->value();
```
Comment (bastianallgeier) [inline src/Cms/System.php]: It's probably never going to be performance-relevant, but I would still improve it a bit. 

```suggestion
        $site = $this->app->site();

        if ($site->title()->isNotEmpty()) {
          return $site->title()->value();
        }
        
        return $site->blueprint()->title();
```

**Review by bastianallgeier — COMMENTED** [review]
The rest looks good to me

**Comment by distantnative** [comment]
👍

**Review by bastianallgeier — DISMISSED** [review]

**Review by bastianallgeier — DISMISSED** [review]

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**